### PR TITLE
Fix delphi compilation hints

### PR DESCRIPTION
**Pull request by reyandme**

### CODE REVIEW

File: src/gui/pages_menu/KM_GUIMenuMapEditor.pas [hunk 1]
```diff
@@ -438,6 +438,7 @@ procedure TKMMenuMapEditor.ColumnClick(aValue: Integer);
 
 function TKMMenuMapEditor.GetMaps: TKMapsCollection;
 begin
+  Result := nil;
```
Comment (Kromster80) [inline src/gui/pages_menu/KM_GUIMenuMapEditor.pas]: I came to a better pattern:

case of
  1:
  2:
else
  raise Exception.Create();
end;

It keeps compiler happy, takes less LOC and catches errors well.

**Review by Kromster80 — CHANGES_REQUESTED** [review]
Can you use this pattern?

**Comment by reyandme** [comment]
I find your solution is much better. I refactor all places where Assert(False) occurs.
And at least 4 times I find places where there was extra unused code. Compiler get more info about code when we use Exception.Create and can get some hints because of that.

There is still one last hint: unused private method:

`procedure TKMNetServer.SendMessageToRoom(aKind: TKMessageKind; aRoom: Integer; aParam: Integer);`

Can we delete it ?

**Comment by lewinjh** [comment]
Yes it should be safe to delete SendMessageToRoom, I think it's a leftover from when we first implemented multiplayer. You only ever communicate to your own room, which is enforced by the server.

**Comment by reyandme** [comment]
ok, I've deleted it

**Pull request merged**